### PR TITLE
Support multi-plugin projects

### PR DESCRIPTION
**Pull request by devinivy**
See #8 for details

### CODE REVIEW

File: test/index.js [hunk 1]
```diff
@@ -157,13 +157,49 @@ describe('hpal', () => {
...
+                    })
+                    .then((contents) => {
+
+                        expect(contents).to.startWith('\'use strict\';');
+
+                        return rimraf('ambiguous-hc-file/project-a/routes');
+                    });
+            });
+
+            it('succeeds when finding a .hc.js file from a cwd deep in the project.', () => {
```
Comment (zemccartney) [inline test/index.js]: oh woah, that's awesome! 👏

File: test/index.js [hunk 2]
```diff
@@ -157,13 +157,49 @@ describe('hpal', () => {
...
+
+                        expect(result.err).to.not.exist();
+                        expect(result.output).to.contain('Wrote ../project-b/routes/index.js');
+                        expect(result.errorOutput).to.equal('');
+
+                        return read('non-ambiguous-hc-file-cwd/project-b/routes/index.js');
+                    })
+                    .then((contents) => {
+
+                        expect(contents).to.startWith('\'use strict\';');
```
Comment (zemccartney) [inline test/index.js]: couple of things:

1. the `read` helper broke my brain a little bit. so ill!
2. this caught my eye; why assert against "use strict"?  I think curious b/c this read as not specific to the file created, so I guess not descriptive (to human reader me :) ) of what success looks like, but also figure that's a super surface-level thought

**Comment by zemccartney** [comment]
this looks super cool. thanks for thinking about this and drumming this up. 🙏

File: test/index.js (hunk 2 above)
Comment (devinivy) [reply test/index.js]: That's a good point.  It is a really superficial check, and I can see the possibility for false positives as you pointed out.  The thing is, hpal is only responsible for a small amount of the contents of this file, and the rest comes from haute-couture, so it's pretty hard to fingerprint these files.  I will think about it though! Thanks for the thought 🙏

Maybe we could ensure the file does not exist before running the command :)

**Comment by devinivy** [comment]
Thanks for taking a good look at this @zemccartney !

As discussed off github, we may want to take a slightly different approach to handle a case like this one,
```
lib/
    .hc.js
    plugins/          <-- cwd
        sub-plugin/
            .hc.js
```

Under the old behavior it would correctly (I think) choose `lib/` as our working plugin.  Under the behavior in this PR it would start to choose `sub-plugin/`.  Will be thinking on it!

File: test/index.js (hunk 2 above)
Comment (zemccartney) [reply test/index.js]: oh very cool, yea, didn't think about the existence check at all
👍 mm, all sounds good. thanks for that

**Comment by devinivy** [comment]
Replaced by #10.

**Pull request closed without merging**